### PR TITLE
WIP: 221 implementation api

### PR DESCRIPTION
**Pull request by nymann**

### CODE REVIEW

File: project_report/tex/implementation.tex [hunk 1]
```diff
@@ -24,6 +24,24 @@ \subsection{Dashboard}
 uses two separate libraries for making graphs and getting icons for the buttons.
 
 \subsection{REST API}
+The REST API is responsible for providing the dashboard with live data, and the
+ability to start new batches from the dashboard. The API is able to provide
+live data due to the OPC-UA client posting it to the API as soon as a property
+changes, the data is then kept in the PostgreSQL database. The API is developed
+in Python3 via the popular FastAPI framework. For interacting with the database
+it's using the async ORM Gino. The source code structure is inspired from best
```
Comment (Kechr) [inline project_report/tex/implementation.tex]: No contractions please: it's -> it is

File: project_report/tex/implementation.tex [hunk 2]
```diff
@@ -24,6 +24,24 @@ \subsection{Dashboard}
...
+practices in MVC, with separations between the different layers (router,
+service, schemas and database models (ORM)). This ensures that components can
+easily be swapped out, and thereby makes it easier to maintain in the long run.
+Another part of implementation is the development setup, via our continous
+integration pipeline we check if the code is linted properly as well as unit
+and system tests are passing, if one of these steps are failing the pull request
+is not allowed to be merged to master. When new branches are merged to master,
+the last part of our pipeline is being run which builds a new docker image to
+our private registry which is then deployed automatically to 
+https://api.bierproductie.nymann.dev/docs, this can be verified via the version
```
Comment (Kechr) [inline project_report/tex/implementation.tex]: \url{weburl}

File: project_report/tex/implementation.tex [hunk 3]
```diff
@@ -24,6 +24,24 @@ \subsection{Dashboard}
...
+The REST API is responsible for providing the dashboard with live data, and the
+ability to start new batches from the dashboard. The API is able to provide
+live data due to the OPC-UA client posting it to the API as soon as a property
+changes, the data is then kept in the PostgreSQL database. The API is developed
+in Python3 via the popular FastAPI framework. For interacting with the database
+it's using the async ORM Gino. The source code structure is inspired from best
+practices in MVC, with separations between the different layers (router,
+service, schemas and database models (ORM)). This ensures that components can
+easily be swapped out, and thereby makes it easier to maintain in the long run.
+Another part of implementation is the development setup, via our continous
```
Comment (Kechr) [inline project_report/tex/implementation.tex]: Personal pronoun. Please rephrase.

Comment (nymann) [reply project_report/tex/implementation.tex]: Fixed

File: project_report/tex/implementation.tex (hunk 1 above)
Comment (nymann) [reply project_report/tex/implementation.tex]: Fixed

**Pull request merged**